sys/kern/syscalls.c

```diff
@@ -1330,5 +1330,32 @@ static int sys_kevent(proc_t *p, kevent_args_t *args, register_t *res) {
 
 static int sys_sigtimedwait(proc_t *p, sigtimedwait_args_t *args,
                             register_t *res) {
-  return ENOTSUP;
+  const sigset_t *u_set = SCARG(args, set);
+  siginfo_t *u_info = SCARG(args, info);
+  const timespec_t *u_timeout = SCARG(args, timeout);
+  sigset_t set;
+  ksiginfo_t ksi;
+  timespec_t timeout = {};
+  int error;
+
+  if (u_timeout) {
+    error = copyin_s(u_timeout, timeout);
+    if (error)
+      return error;
+  }
+
+  error = copyin_s(u_set, set);
+  if (error)
+    return error;
+
+  error = do_sigtimedwait(p, set, &ksi, u_timeout ? &timeout : NULL);
+  if (error)
+    return error;
+
+  if (u_info) {
+    error = copyout_s(ksi.ksi_info, u_info);
+  }
```

Comment:
You are not consistent with style. { }

Reply:
Right. Which one is preferable in our project?

Reply:
For single line if-clause we don't use `{ }`. If in doubt look at surrounding code and see what's more prevalent.